### PR TITLE
Week9: domain model and a new app feature

### PR DESCRIPTION
**Pull request by nlpersimon**

### CODE REVIEW

File: app/controllers/app.rb [hunk 1]
```diff
@@ -31,14 +31,73 @@ class App < Roda
...
+            #   skills.append(Skiller::Entity::Skill.new(id:id, name:name, job_db_id:job_db_id, salary:salary))
+            # end
+
+            ## TODO: then use `Repository::JobsSkills.create()` to put them into database
+            # Repository::JobsSkills.create(skills)
+
+            ## TODO: use `Repository::QueriesJobs.find_skills_by_query()` if the query has been searched
+            # skills.each do |skill|
+            #  puts skill.name
+            # end
```
Comment (tienyulin) [inline app/controllers/app.rb:69]: Please remove these commented-out code. Only Todo is acceptable.

File: app/controllers/app.rb [hunk 2]
```diff
@@ -31,14 +31,73 @@ class App < Roda
...
+        Repository::QueriesJobs.create(query, jobs.map(&:db_id))
+        jobs
+      end
+
+      def request_first_10_full_jobs(query)
+        partial_jobs = @job_mapper.job_list(query)
+        partial_jobs[0...10].map { |pj| @job_mapper.job(pj.job_id) }
+      end
+    end
     # puts config.DB_FILENAME
```
Comment (tienyulin) [inline app/controllers/app.rb:101]: Please remove this commented-out code.

File: app/domain/skills/mappers/skill_mapper.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# require_relative '../../../infrastructure/skill_extractor/init'
```
Comment (tienyulin) [inline app/domain/skills/mappers/skill_mapper.rb:3]: Please remove this commented-out code.

File: app/infrastructure/database/migrations/001_jobs_create.rb [hunk 4]
```diff
@@ -6,16 +6,17 @@
...
-      String      :job_id, unique: true
+      Integer     :job_id, unique: true
       String      :job_title
       String      :description
       String      :location
       Float       :min_year_salary
       Float       :max_year_salary
       String      :currency
       String      :url, null: true
+      Bool        :isfull
```
Comment (tienyulin) [inline app/infrastructure/database/migrations/001_jobs_create.rb:19]: Highly recommend following the snake case. `is_full`

File: app/infrastructure/database/orm/job_skill_orm.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# http://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/Timestamps.html
```
Comment (tienyulin) [inline app/infrastructure/database/orm/job_skill_orm.rb:3]: Please remove this commented-out code.

File: app/infrastructure/database/orm/query_job_orm.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# http://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/Timestamps.html
```
Comment (tienyulin) [inline app/infrastructure/database/orm/query_job_orm.rb:3]: Please remove this commented-out code.

**Review by tienyulin — COMMENTED** [review]
HW9, reviewed.

File: app/controllers/app.rb (hunk 1 above)
Comment (nlpersimon) [reply app/controllers/app.rb:69]: Issue solved. Please refer to these lines of the [commit](https://github.com/KeroKnows/s-killer/commit/c1d43b9257897b20edca40169e25f234a0879b85#diff-31c5cc96aa2344aa33c91fea07835f0d203de1f6b24a29b215baf9a96de28607L43).

File: app/controllers/app.rb (hunk 2 above)
Comment (nlpersimon) [reply app/controllers/app.rb:101]: Issue solved. Please refer to this line of the [commit](https://github.com/KeroKnows/s-killer/commit/c1d43b9257897b20edca40169e25f234a0879b85#diff-31c5cc96aa2344aa33c91fea07835f0d203de1f6b24a29b215baf9a96de28607L101).

File: app/domain/skills/mappers/skill_mapper.rb (hunk 3 above)
Comment (nlpersimon) [reply app/domain/skills/mappers/skill_mapper.rb:3]: Issue solved. Please refer to this line of the [commit](https://github.com/KeroKnows/s-killer/commit/c1d43b9257897b20edca40169e25f234a0879b85#diff-adc1eee32180bfa35f4ca82ddfbdb72d99134ebe927a7f0fdb9117ed3bc2b354L3).

File: app/infrastructure/database/orm/job_skill_orm.rb (hunk 5 above)
Comment (nlpersimon) [reply app/infrastructure/database/orm/job_skill_orm.rb:3]: Issue solved. Please refer to this line of the [commit](https://github.com/KeroKnows/s-killer/commit/c1d43b9257897b20edca40169e25f234a0879b85#diff-cdc2ae7dba7afb211ecd7c0f113e19ae424730b47e9d212d9b5cf109fd8cce6eL3).

File: app/infrastructure/database/orm/query_job_orm.rb (hunk 6 above)
Comment (nlpersimon) [reply app/infrastructure/database/orm/query_job_orm.rb:3]: Issue solved. Please refer to this line of the [commit](https://github.com/KeroKnows/s-killer/commit/c1d43b9257897b20edca40169e25f234a0879b85#diff-035149020368d3a3eefdf2929c3f5b5054917c0d2bc787a0e7061a2f3bdb6014L3).

File: app/infrastructure/database/migrations/001_jobs_create.rb (hunk 4 above)
Comment (nlpersimon) [reply app/infrastructure/database/migrations/001_jobs_create.rb:19]: Issue solved. Please refer to this [commit](https://github.com/KeroKnows/s-killer/commit/6df43afb8140f4ee8e82fc5f9e29205f984a01ce).
Also, you need to re-migrate the database if you would like to run the tests.
Thank you for all the work again!

**Pull request merged**